test/controllers/who.is.making.filing.controller.spec.ts

```diff
@@ -128,4 +167,51 @@ describe("Who is making filing controller tests", () => {
       expect(resp.text).toContain(SERVICE_UNAVAILABLE);
     });
   });
+
+  describe("POST with url params tests", () => {
+    test(`redirect the ${config.OVERSEAS_ENTITY_DUE_DILIGENCE_URL} page when ${WhoIsRegisteringType.SOMEONE_ELSE} is selected`, async () => {
+      const resp = await request(app)
+        .post(config.WHO_IS_MAKING_FILING_WITH_PARAMS_URL)
+        .send({ [WhoIsRegisteringKey]: WhoIsRegisteringType.SOMEONE_ELSE });
+
+      expect(resp.status).toEqual(302);
+      expect(resp.header.location).toEqual(config.OVERSEAS_ENTITY_DUE_DILIGENCE_URL);
+      expect(mockSetExtraData).toHaveBeenCalledTimes(1);
```

Comment:
You have missed assertions here that need adding. These will then bring in further test dependencies/imports that you'll need to set up.

Please have another look at the changes we did for the presenter URLs and the unit test changes done for the POST calls:

https://github.com/companieshouse/overseas-entities-web/pull/1099/files#diff-486722c6890797dc94b8e988d01d514539ef8627a512c8ae5a2acc03780d8ffc

Reply:
Updated the tests but the post oe due diligence seems to keep finding only the due diligence url for some reason and failing. Commented the line out for the time being, cannot find what it is that's doing this.

Reply:
On line 190 of the test the return value is mocked but seems to be ignored by the test. Added the extra assertion.